### PR TITLE
Fix : event catched as many time as client created

### PR DESCRIPTION
**Pull request by mojoLyon**

### CODE REVIEW

**Comment by omansour** [comment]
:+1:

File: tests/Units/DependencyInjection/M6WebGuzzleHttpExtension.php [hunk 1]
```diff
@@ -161,6 +162,30 @@ public function testEventDispatcherMiddleWare()
...
+            ->and($promises = [
+                'test' => $client->getAsync('http://httpbin.org'),
+                'test2' => $client->getAsync('http://httpbin.org/ip')
+            ])
+            ->and($rep = Promise\unwrap($promises))
+            ->and($client2->get('http://httpbin.org'))
+            ->then
+                ->mock($mockDispatcher)
+                ->call('dispatch')
+                    ->exactly(3)
```
Comment (mikaelrandy) [inline tests/Units/DependencyInjection/M6WebGuzzleHttpExtension.php]: Pourquoi 3 appels ?

Comment (omansour) [inline tests/Units/DependencyInjection/M6WebGuzzleHttpExtension.php]: parceque 3 urls sur 2 clients ?

Comment (mikaelrandy) [inline tests/Units/DependencyInjection/M6WebGuzzleHttpExtension.php]: Ah oui, j'avais raté la ligne 181 !

**Comment by mikaelrandy** [comment]
:+1:

**Pull request merged**